use-cases/meeting-scheduling/src/main/java/org/acme/meetingschedule/domain/MeetingAssignment.java

```diff
@@ -0,0 +1,152 @@
+import ai.timefold.solver.core.api.domain.entity.PlanningPin;
+import ai.timefold.solver.core.api.domain.lookup.PlanningId;
+import ai.timefold.solver.core.api.domain.variable.PlanningVariable;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+@PlanningEntity
+public class MeetingAssignment {
+
+    private String id;
+    private Meeting meeting;
+    private boolean pinned;
+
+    // Planning variables: changes during planning, between score calculations.
+    private TimeGrain startingTimeGrain;
+    private Room room;
+
+    public MeetingAssignment() {
+    }
+
+    public MeetingAssignment(String id) {
+        this.id = id;
+    }
+
+    public MeetingAssignment(String id, Meeting meeting) {
+        this(id);
+        this.meeting = meeting;
+    }
+
+    public MeetingAssignment(String id, Meeting meeting, TimeGrain startingTimeGrain, Room room) {
+        this(id, meeting);
+        this.startingTimeGrain = startingTimeGrain;
+        this.room = room;
+    }
+
+    @PlanningId
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public Meeting getMeeting() {
+        return meeting;
+    }
+
+    public void setMeeting(Meeting meeting) {
+        this.meeting = meeting;
+    }
+
+    @PlanningPin
+    public boolean isPinned() {
+        return pinned;
+    }
+
+    public void setPinned(boolean pinned) {
+        this.pinned = pinned;
+    }
+
+    @PlanningVariable
+    public TimeGrain getStartingTimeGrain() {
+        return startingTimeGrain;
+    }
+
+    public void setStartingTimeGrain(TimeGrain startingTimeGrain) {
+        this.startingTimeGrain = startingTimeGrain;
+    }
+
+    @PlanningVariable
+    public Room getRoom() {
+        return room;
+    }
+
+    public void setRoom(Room room) {
+        this.room = room;
+    }
+
+    // ************************************************************************
+    // Complex methods
+    // ************************************************************************
+
+    @JsonIgnore
+    public int getGrainIndex() {
+        return getStartingTimeGrain().getGrainIndex();
+    }
+
+    @JsonIgnore
+    public int calculateOverlap(MeetingAssignment other) {
+        if (startingTimeGrain == null || other.getStartingTimeGrain() == null) {
+            return 0;
+        }
+        // start is inclusive, end is exclusive
+        int start = startingTimeGrain.getGrainIndex();
+        int otherStart = other.startingTimeGrain.getGrainIndex();
+        int otherEnd = otherStart + other.meeting.getDurationInGrains();
+        if (otherEnd < start) {
+            return 0;
+        }
+        int end = start + meeting.getDurationInGrains();
+        if (end < otherStart) {
+            return 0;
+        }
+        return Math.min(end, otherEnd) - Math.max(start, otherStart);
+    }
+
+    @JsonIgnore
+    public Integer getLastTimeGrainIndex() {
```

Comment:
Arguably if the counterpart to this is called `getStarting...`, this one should be `getEnding...`. Or they should be first/last. But it should be mutually consistent.